python/semantic_kernel/connectors/ai/mistral_ai/services/mistral_ai_chat_completion.py

```diff
@@ -116,8 +125,64 @@ async def get_chat_message_contents(
 
         if not settings.ai_model_id:
             settings.ai_model_id = self.ai_model_id
+        
+        kernel = kwargs.get("kernel", None)
+        if settings.function_choice_behavior is not None and kernel is None:
+            raise ServiceInvalidExecutionSettingsError("The kernel is required for MistralAI tool calls.")
+        assert isinstance(kernel, Kernel)  # nosec
+            
+        self._update_settings(settings, chat_history, kernel=kernel)
+
+        # behavior for non-function calling or for enable, but not auto-invoke.
+        if settings.function_choice_behavior is None or (
+            settings.function_choice_behavior and not settings.function_choice_behavior.auto_invoke_kernel_functions
+        ):
+            return await self._send_chat_request(settings)
+        
+        # loop for auto-invoke function calls
+        for request_index in range(settings.function_choice_behavior.maximum_auto_invoke_attempts):
+            completions = await self._send_chat_request(settings)
+            # there is only one chat message, this was checked earlier
+            chat_history.add_message(message=completions[0])
```

Comment:
Please help update this pattern based on #8300. The chat history message shouldn't be added here, but before line 152 if function calls exist. Additionally, this should be:

```python
function_calls = [item for item in completions[0].items if isinstance(item, FunctionCallContent)]
```